cg/meta/workflow/mutant/quality_controller/quality_controller.py

```diff
@@ -0,0 +1,174 @@
+            report_file_path=report_file_path,
+        )
+
+        return QualityResult(
+            case_quality_result=case_quality_result,
+            samples_quality_results=samples_quality_results,
+            summary=summary,
+        )
+
+    def quality_control_samples(self, quality_metrics: QualityMetrics) -> SamplesQualityResults:
+        samples_quality_results: list[SampleQualityResults] = []
+        for sample in quality_metrics.pool.samples:
+            sample_results: SampleResults = quality_metrics.results[sample.internal_id]
+            sample_quality_results: SampleQualityResults = self.quality_control_sample(
+                sample=sample, sample_results=sample_results
+            )
+            samples_quality_results.append(sample_quality_results)
+
+        internal_negative_control_sample: Sample = quality_metrics.pool.internal_negative_control
+        internal_negative_control_sample_results: SampleResults = quality_metrics.results[
+            internal_negative_control_sample.internal_id
+        ]
+        internal_negative_control_quality_metrics: (
+            SampleQualityResults
+        ) = self.quality_control_sample(
+            sample=internal_negative_control_sample,
+            sample_results=internal_negative_control_sample_results,
+        )
+
+        external_negative_control_sample: Sample = quality_metrics.pool.external_negative_control
+        external_negative_control_sample_results: SampleResults = quality_metrics.results[
+            external_negative_control_sample.internal_id
+        ]
+        external_negative_control_quality_metrics: (
+            SampleQualityResults
+        ) = self.quality_control_sample(
+            sample=external_negative_control_sample,
+            sample_results=external_negative_control_sample_results,
+        )
+
+        return SamplesQualityResults(
+            samples=samples_quality_results,
+            internal_negative_control=internal_negative_control_quality_metrics,
+            external_negative_control=external_negative_control_quality_metrics,
+        )
+
+    def quality_control_sample(
+        self,
+        sample: Sample,
+        sample_results: SampleResults,
+        external_negative_control: bool = False,
+        internal_negative_control: bool = False,
+    ) -> SampleQualityResults:
+        sample_has_valid_total_reads: bool = has_valid_total_reads(
+            sample=sample,
+            external_negative_control=external_negative_control,
+            internal_negative_control=internal_negative_control,
+        )
+        sample_passes_qc: bool = sample_has_valid_total_reads and sample_results.qc_pass
+
+        sample_quality = SampleQualityResults(
+            sample_id=sample.internal_id,
+            passes_qc=sample_passes_qc,
+            passes_reads_threshold=sample_has_valid_total_reads,
+            passes_mutant_qc=sample_results.qc_pass,
+        )
+
```

Comment:
Do we care if external_negative_control passes/fails mutant qc?

Reply:
Yes!
If it passes mutantQC, the QC should fail. Then reason should be stated in the QC report.